lib/src/indicator.dart

```diff
@@ -42,52 +47,118 @@ class _PageIndicatorState extends State<PageIndicator> {
 
   @override
   Widget build(BuildContext context) {
+    CustomPainter indicatorPainter;
+    switch (widget.indicatorShape) {
+      case IndicatorShape.roundRectangle:
+        indicatorPainter = RRectPainter(
+            color: widget.color,
+            selectedColor: widget.selectedColor,
+            count: widget.length,
+            page: widget.controller.page ?? controller.initialPage.toDouble(),
+            padding: widget.indicatorSpace,
+            width: widget.size);
+        break;
+      case IndicatorShape.circle:
+      default:
+        indicatorPainter = CirclePainter(
+            color: widget.color,
+            selectedColor: widget.selectedColor,
+            count: widget.length,
+            page: widget.controller.page ?? controller.initialPage.toDouble(),
+            padding: widget.indicatorSpace,
+            radius: widget.size / 2);
+        break;
+    }
+
     return IgnorePointer(
-      child: CustomPaint(
-        child: Container(
-          height: double.infinity,
-        ),
-        painter: CirclePainter(
-          color: widget.color,
-          selectetColor: widget.selectedColor,
-          count: widget.length,
-          page: widget.controller.page ?? controller.initialPage.toDouble(),
-          padding: widget.indicatorSpace,
-          radius: widget.size / 2,
-        ),
+        child: CustomPaint(
+      child: Container(
+        height: double.infinity,
       ),
-    );
+      painter: indicatorPainter,
+    ));
   }
 }
 
+class RRectPainter extends CustomPainter {
+  double page;
+  int count;
+  Color color;
+  Color selectedColor;
+  double width;
+  double padding;
+  Paint _circlePaint;
+  Paint _selectedPaint;
+  RRectPainter(
+      {this.page = 0.0,
+      this.count = 0,
+      this.color = Colors.white,
+      this.selectedColor = Colors.grey,
+      this.width = 12.0,
+      this.padding = 5.0}) {
+    _circlePaint = Paint();
+    _circlePaint.color = color;
+
+    _selectedPaint = Paint();
+    _selectedPaint.color = selectedColor;
+
+    this.page ??= 0.0;
+    this.count ??= 0;
+    this.color ??= Colors.white;
+    this.selectedColor ??= Colors.grey;
+    this.width ??= 6.0;
+    this.padding ??= 5.0;
+  }
+double get totalWidth => count * width + padding * (count - 1);
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    var height = width / 3;
+    var centerWidth = size.width / 2;
+    var startX = centerWidth - totalWidth / 2;
+    for (var i = 0; i < count ?? 0; i++) {
+      var x = startX + i * (width + padding);
+      var rect = Rect.fromLTWH(x, 0, width, height);
```

Comment:
I think y here should be changed according to the current align.
Otherwise, when padding is 0, I change align to top get image like the:
![image](https://user-images.githubusercontent.com/14145407/56262742-9e9e7080-6112-11e9-9a8b-926ba1d0c428.png)

When align is a bottom, it's like this:

![image](https://user-images.githubusercontent.com/14145407/56262826-f89f3600-6112-11e9-8233-fd9c9b71aefd.png)

Reply:
Ok...

Reply:
Was just thinking, Shouldn't the top padding be handled by the end user via the container he/she is using?

Reply:
If you want to separate indicator and PageView, then the user handles the location of Indicator himself.
You need to create a Controller yourself and set the Controller to Indicator and PageView.
But for users, although more flexible, they need more code.

Less customization is sometimes not a bad thing. 

If you need too many customization options or even strong relevance to business logic, it's better to consider your own definitions.

Reply:
I checked it and found that issue exists in circular as well, Should I make changes there too...

Reply:
As long as the actual use conforms to the meaning of `padding`.
😄

Reply:
I have added top padding to the default padding.. Please validate